api/src/main/java/org/itsallcode/openfasttrace/api/core/LinkedSpecificationItem.java

```diff
@@ -208,7 +245,7 @@ public void addLinkToItemWithStatus(final LinkedSpecificationItem item, final Li
      */
     public List<SpecificationItemId> getCoveredIds()
     {
-        return this.getItem().getCoveredIds();
+        return List.copyOf(this.getItem().getCoveredIds());
```

Comment:
Why copy?

Reply:
Because the list is used often in a loop (in many of your tests) that again calls a method in LinkedSpecificationItem which changes (adds I think) items in the list therefore producing an Exception. As the caller does just should be able to call methods of LinkedSpecificationItem without hesitating that some concurrent access exception happen (as the existing tests did) I returned a copy here.